lldb/source/Core/Debugger.cpp

```diff
@@ -1425,10 +1426,19 @@ void Debugger::SetLoggingCallback(lldb::LogOutputCallback log_callback,
       std::make_shared<CallbackLogHandler>(log_callback, baton);
 }
 
+void Debugger::AddDestroyCallback(
+    lldb_private::DebuggerDestroyCallback destroy_callback, void *baton) {
+  m_destroy_callback_and_baton.emplace_back(destroy_callback, baton);
```

Comment:
You need some locking here.  Since there's an SB API and an internal way to add destroy callbacks, you can't guarantee there won't be concurrent add operations.

Reply:
Good to know about the potential concurrency. Thanks!

Reply:
Addressed in the latest version.